jolt-inlines/sha2/src/trace_generator.rs

```diff
@@ -446,75 +450,37 @@ impl Sha256SequenceBuilder {
-enum Value {
-    Imm(u64),
-    Reg(u8),
+const VIRTUAL_REGISTER_COUNT: u64 = 32; //  see Section 6.1 of Jolt paper
+pub const fn virtual_register_index(index: u64) -> u64 {
+    index + VIRTUAL_REGISTER_COUNT
 }
 
-pub fn execute_sha256_compression_initial(input: [u32; 16]) -> [u32; 8] {
-    execute_sha256_compression(BLOCK.map(|x| x as u32), input)
+// Virtual instructions builder for sha256 - returns empty sequence as XOR is atomic
+pub fn sha2_virtual_sequence_builder(_address: u64, _rs1: u8, _rs2: u8) -> Vec<RV32IMInstruction> {
+    // Virtual registers used as a scratch space
+    let mut vr = [0u8; 32];
+    (0..32).for_each(|i| {
+        vr[i] = virtual_register_index(i as u64) as u8;
+    });
+    let builder = Sha256SequenceBuilder::new(
+        _address, vr, _rs1, _rs2, false, // not initial - uses custom IV from rs2
+    );
+    builder.build()
 }
 
-pub fn execute_sha256_compression(initial_state: [u32; 8], input: [u32; 16]) -> [u32; 8] {
-    let mut a = initial_state[0];
-    let mut b = initial_state[1];
-    let mut c = initial_state[2];
-    let mut d = initial_state[3];
-    let mut e = initial_state[4];
-    let mut f = initial_state[5];
-    let mut g = initial_state[6];
-    let mut h = initial_state[7];
-
-    let mut w = [0u32; 64];
-
-    w[..16].copy_from_slice(&input);
-
-    // Calculate word schedule
-    for i in 16..64 {
-        // σ₁(w[i-2]) + w[i-7] + σ₀(w[i-15]) + w[i-16]
-        let s0 = w[i - 15].rotate_right(7) ^ w[i - 15].rotate_right(18) ^ (w[i - 15] >> 3);
-        let s1 = w[i - 2].rotate_right(17) ^ w[i - 2].rotate_right(19) ^ (w[i - 2] >> 10);
-        w[i] = w[i - 16]
-            .wrapping_add(s0)
-            .wrapping_add(w[i - 7])
-            .wrapping_add(s1);
-    }
-
-    // Perform 64 rounds
-    for i in 0..64 {
-        let ch = (e & f) ^ ((!e) & g);
-        let maj = (a & b) ^ (a & c) ^ (b & c);
-
-        let sigma0 = a.rotate_right(2) ^ a.rotate_right(13) ^ a.rotate_right(22); // Σ₀(a)
-        let sigma1 = e.rotate_right(6) ^ e.rotate_right(11) ^ e.rotate_right(25); // Σ₁(e)
-
-        let t1 = h
-            .wrapping_add(sigma1)
-            .wrapping_add(ch)
-            .wrapping_add(K[i] as u32)
-            .wrapping_add(w[i]);
-        let t2 = sigma0.wrapping_add(maj);
-
-        h = g;
-        g = f;
-        f = e;
-        e = d.wrapping_add(t1);
-        d = c;
-        c = b;
-        b = a;
-        a = t1.wrapping_add(t2);
-    }
-
-    // Final IV addition
-    [
-        initial_state[0].wrapping_add(a),
-        initial_state[1].wrapping_add(b),
-        initial_state[2].wrapping_add(c),
-        initial_state[3].wrapping_add(d),
-        initial_state[4].wrapping_add(e),
-        initial_state[5].wrapping_add(f),
-        initial_state[6].wrapping_add(g),
-        initial_state[7].wrapping_add(h),
-    ]
+// Virtual instructions builder for sha256_init - returns empty sequence as XOR is atomic
+pub fn sha2_init_virtual_sequence_builder(
+    _address: u64,
```

Comment:
It seems like these variables are used in the function, but prefixed with `_` as unused variables.